src/modules/ChannelList/context/ChannelListProvider.tsx

```diff
@@ -208,6 +210,26 @@ const ChannelListProvider: React.FC<ChannelListProviderProps> = (props: ChannelL
         markAsDeliveredScheduler,
         disableMarkAsDelivered,
       });
+    },
+    [
+      channelListDispatcher,
+      sdk,
+      logger,
+      userFilledChannelListQuery,
+      sortChannelList,
+      setChannelSource,
+      disableAutoSelect,
+      disableMarkAsDelivered,
+      markAsDeliveredScheduler,
+      onChannelSelect,
+    ]
+  );
+
+  useEffect(() => {
+    const sdkChannelHandlerId = uuidv4();
+    onSetupChannelList(sdkChannelHandlerId);
```

Comment:
Shouldn't `onSetupChannelList` be within the `if (sdkIntialized) {` block?

Reply:
Good catch!